src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -331,7 +333,8 @@ private static void collectDeclarations(Deque<AbstractFrame> frameStack, DetailA
                 collectVariableDeclarations(ast, frame);
                 break;
             case TokenTypes.PARAMETER_DEF :
-                if (!CheckUtils.isReceiverParameter(ast)) {
```

Comment:
@romani 
The diff report will never be able to generate many differences. The reason is:

When lambdas in java files are parsed and the ast is created, if the lambda has multiple arguments, i.e arguments like (a,b) and so on then they are treated as parameter definitions.
Since they are treated as parameter definitions beginTree and collectDeclarations end up collecting lambda parameters as local variables
Since local variables are not to throw any violations unless overlapping (issue #3423  fixed) there won't be any violations in the master for them
But my patch actually stops them from getting collected at all and hence again there won't be any violation when there are multiple lambda parameters